autoload/neomake.vim

```diff
@@ -421,11 +444,20 @@ function! neomake#MakeHandler(job_id, data, event_type) abort
     endif
 endfunction
 
-function! s:CleanSignsAndErrors()
-    call neomake#signs#CleanOldSigns()
-    if !s:cleared_current_errors
-        let s:current_errors = {}
-        let s:cleared_current_errors = 1
+function! s:CleanAllSignsAndErrors()
+    for bufnr in keys(s:cleared_current_errors)
+        call s:CleanSignsAndErrors(bufnr)
+    endfor
+endfunction
+
+function! s:CleanSignsAndErrors(bufnr)
+    call neomake#signs#CleanOldSigns(a:bufnr)
+    if has_key(s:cleared_current_errors, a:bufnr) && !s:cleared_current_errors[a:bufnr]
```

Comment:
Not sure we need this `s:cleared_current_errors` variable. It seems like if `s:current_errors` doesn't have the `a:bufnr` key, then clearing the errors is essentially a no-op anyway.